api/app/src/models/medical/contact.ts

```diff
@@ -1,3 +1,9 @@
 export type ContactTypes = "email" | "phone";
 
-export type Contact = Partial<Record<ContactTypes, string | undefined>>;
+export type Contact =
+  | {
+      email?: string | null;
+      phone?: string | null;
+    }
+  | null
+  | undefined;
```

Comment:
Using `Record` makes it safer since it will require that we keep instances of `Contact` in sync with `ContactTypes`. This new version allows one to add a `ContactType` and to forget to update `Contact`.